src/simulation_lifecycle/simulation.c

```diff
@@ -69,7 +73,7 @@ int parse_default_sim_config(const cJSON *simulation_config, cJSON *target) {
     }
     // Check that the model exists
     char *model_path = concat(SIM_MODEL_LIBRARY, cJSON_GetStringValue(model));
-    int model_found = executable_exists(model_path);
+    int model_found = file_exists(model_path);
```

Comment:
Why did you change this? The executable file needs execution permissions, not read permissions. Check the difference between the functions `executable_exists` and `file_exists`

Reply:
It would not work but I'll check again the difference between the functions.

Reply:
I've made the change at both places. It is working as expected.